### ISSUE TITLE
safeTry with async generator should return ResultAsync

### ISSUE THREAD

**Issue by mattpocock** [supermacro/neverthrow#561]
```ts
import { safeTry, ok } from "neverthrow";

safeTry(async function* () {
  return ok("Hey");
}).map((str) => str.toUpperCase());
```

This should be possible, but isn't.

This makes the lib less composable.

**Comment by supermacro** [comment]
Fixed by #562

**Comment by mattpocock** [comment]
Siiiiiick I'm really excited about this.
